### ISSUE TITLE
support react native stylesheet (css on js)

### ISSUE THREAD

**Issue by enapupe** [eslint/eslint#6562]
I'm starting a `react-native` project and the CSS configuration for UI is set in a JS object such as
``` 
import {StyleSheet} from 'react-native'

const styles = StyleSheet.create({
  applicationView: {
    flex: 1
  }
  , container: {
    flex: 1
    , justifyContent: 'center'
    , backgroundColor: '#fff'
  }
  , welcome: {
    fontSize: 20
    , textAlign: 'center'
    , margin: 20
  }
  , myImage: {
    width: 200
    , height: 200
    , alignSelf: 'center'
  }
})

export default styles
```

The issue is that I'm running `no-magic-numbers` with this config
```
   "no-magic-numbers": [
      1,
      {
        "ignore": [
          -1,
          0,
          1,
          2
        ],
        "detectObjects": true,
        "enforceConst": true
      }
    ],
```

This is triggering a lot of warnings. Is there any feature on the roadmap providing some kind of ignore pattern or support for css in js? I really don't want to add eslint comments in every css file.
Adding css files to ignore is also not an option since I need most of its validations..

**Comment by platinumazure** [comment]
I don't think there's anything on the roadmap specific to your use case. In general, we don't make changes catering to individual libraries/frameworks.

You can use inline configuration comments to disable the no-magic-numbers rule around your stylesheets:

```js
import {StyleSheet} from 'react-native'

/* eslint-disable no-magic-numbers */
const styles = StyleSheet.create({
  applicationView: {
    flex: 1
  }
  , container: {
    flex: 1
    , justifyContent: 'center'
    , backgroundColor: '#fff'
  }
  , welcome: {
    fontSize: 20
    , textAlign: 'center'
    , margin: 20
  }
  , myImage: {
    width: 200
    , height: 200
    , alignSelf: 'center'
  }
})
/* eslint-enable no-magic-numbers */

export default styles
```

**Comment by enapupe** [comment]
It just feels wrong to add such comment to ALL styles files.. That is why I came here, to find a smarter solution, but I understand your POV..

**Comment by ilyavolodin** [comment]
My suggestion would be to either disable `no-magic-numbers` rule all together, or create two different config files, and lint stylesheets in a separate run providing it different config. I don't see us making library specific changes to a global rule, sorry.

**Comment by enapupe** [comment]
I understand it is not under eslint general scope..
I was in touch with some people in `eslint-plugin-react-native` and it seems the ideal approach is a new plugin which extends no-magic-numbers and add more options. Thanks you guys for your attention.
